lib/metasploit/framework/login_scanner/kerberos.rb

```diff
@@ -0,0 +1,93 @@
+require 'metasploit/framework/login_scanner/base'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+
+      # Kerberos User scanner
+      class Kerberos
+        include Metasploit::Framework::LoginScanner::Base
+        include Msf::Exploit::Remote::Kerberos::Client
+
+        DEFAULT_PORT = 88
+
+        def attempt_login(credential)
+          request_options = {
+            credential: credential,
+            host: self.host,
+            port: self.port,
+            protocol: 'tcp',
+            service_name: 'kerberos'
+          }
+
+          begin
+            res = send_request_as(request_options.merge({
+              timeout: self.timeout,
+              client_name: credential.to_h[:username],
```

Comment:
The overhead of `to_h` in a foreach loop looked out of place to me - after a quick search of the code base it looks like everywhere uses `credential.public`; Just confirming if I'm missing anything subtle here 👀